src/codegen/generators/pou_generator.rs

```diff
@@ -699,6 +699,20 @@ impl<'ink, 'cg> PouGenerator<'ink, 'cg> {
                 location.get_column(),
             );
         }
+
+        if function_context.linking_context.is_method()
+            && self.index.find_pou(type_name).is_some_and(|it| it.is_function_block())
+        {
+            let alloca = self.llvm.builder.build_alloca(param_pointer.get_type(), "this");
+            self.llvm.builder.build_store(alloca, param_pointer);
+            index.associate_loaded_local_variable(type_name, "__THIS", alloca)?;
+        }
+        if function_context.linking_context.get_implementation_type() == &ImplementationType::FunctionBlock {
+            let alloca = self.llvm.builder.build_alloca(param_pointer.get_type(), "this");
+            self.llvm.builder.build_store(alloca, param_pointer);
+            index.associate_loaded_local_variable(type_name, "__THIS", alloca)?;
+        }
```

Comment:
I don't think this is actually neccessary because `this` is essentially an alias on the function argument `%0`, no? Specifically I would have assumed `define void @FB_Test(%FB_Test* %0)` to become `define void @FB_Test(%FB_Test* %this)` and then keep the codegen unchanged otherwise. However I wasn't able to find a quick solution for that without panicking so maybe theres another reason? Not a blocker though, can perhaps be improved in another PR

Reply:
we indeed allow `THIS^()`. Leaving `this` out of codegen completely doesnt work because then the variables don't get any association or did I get you wrong @ghaith?
For me what @volsa says makes sense. It just doesn't work for whatever reason :see_no_evil:

Reply:
They should still resolve to the local variable in the POU, so technically no need for this. it's just a marker to resolve the annotation to local instead of global, so the resolving scope during the annotation phase will be local only or similar, and then the annotation will be that of the local variable, when codegen comes along, it finds the annotation of the local variable on the expression and generates that, again theoretical I did not do a deep dive into the implementation.

Reply:
As discussed we leave it as is for now and I created an issue so that we don't forget about this (#1475)